AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/hierarchical-state-machine-swift/StateMachine.swift

```diff
@@ -162,6 +162,16 @@ class StateMachine<
         }
     }
 
+    var currentMachinState: StateType {
```

Comment:
should we do it like this instead `func getCurrentState() async -> StateType`
AND
deprecate `func getCurrentState(_ completion: @escaping (StateType) -> Void)`

Reply:
We want to keep them separate for now since async functions run with different threads than what Dispatch normally uses.